src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -319,11 +319,14 @@ public CommitStats commitStats() {
     /**
      * Read the last segments info from the commit pointed to by the searcher manager
      */
-    protected static SegmentInfos readLastCommittedSegmentInfos(SearcherManager sm) throws IOException {
+    protected static SegmentInfos readLastCommittedSegmentInfos(final SearcherManager sm, final Store store) throws IOException {
         IndexSearcher searcher = sm.acquire();
         try {
             IndexCommit latestCommit = ((DirectoryReader) searcher.getIndexReader()).getIndexCommit();
             return Lucene.readSegmentInfos(latestCommit);
+        } catch (IOException e) {
+            // Fall back to reading from the store if reading from the commit fails
+            return store.readLastCommittedSegmentsInfo();
```

Comment:
can you wrap this in try/catch and add the original exception as suppressed if `readLastCommittedSegmentsInfo` fails?